apps/Test App/Test App/View Models/UploadItemViewModel.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  UploadItemViewModel.swift
+//  Test App
+//
+//  Created by Emily Dixon on 5/16/23.
+//
+
+import Foundation
+import AVFoundation
+import MuxUploadSDK
+
+class UploadItemViewModel: ObservableObject {
+    
+    func startExtractingThumbnail() {
+        guard thumbnailGenerator == nil else {
+            return
+        }
+        
+        thumbnailGenerator = AVAssetImageGenerator(asset: asset)
```

Comment:
This is a seriously cool API!

Reply:
On iOS 16, it even has `async` APIs! I wanted to pay lip service to backward compat tho lol